lib/services/internalEngine/index.js

```diff
@@ -624,4 +570,51 @@ class InternalEngine {
   }
 }
 
+function indexName (index, collection, { internal=true } = {}) {
+  const prefix = internal
+    ? '%'
+    : '#';
+
+  return `${prefix}${index}/${collection}`
+}
+
+function extractIndex (esIndex) {
+  return esIndex.slice(1).split('/')[0];
+}
+
+function extractCollection (esIndex) {
+  return esIndex.slice(1).split('/')[1];
+}
+
+function extractIndexes (esIndexes, { internal=true } = {}) {
+  const indexes = new Set()
+
+  for (const esIndex of esIndexes) {
+    if (! internal && esIndex[0] === '%') {
+      continue;
+    }
+
+    indexes.add(extractIndex(esIndex));
+  }
+
+  return Array.from(indexes);
+}
+
+function extractCollections (esIndexes, index, { internal=true } = {}) {
+  const collections = new Set()
+
+  for (const esIndex of esIndexes) {
+    const [indexName, collectionName] = esIndex.slice(1).split('/');
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "indexName" hides or potentially hides a variable declared in an outer scope at line 573. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)